docs/middleware.rst

```diff
@@ -495,3 +499,101 @@ A legacy transaction still works in the same way as it did before EIP-1559 was i
    ...     'gasPrice': 123456,  # optional - if not provided, gas_price_strategy (if exists) or eth_gasPrice is used
    ... }
    >>> w3.eth.send_transaction(legacy_transaction)
+
+
+Create Middleware
```

Comment:
This seems clear enough but maybe "Creating Middleware"? "Creating Your Own Middleware"?

Reply:
for space consciousness: `Creating Custom Middleware`